testing/tests/calls.rs

```diff
@@ -106,6 +106,76 @@ mod test_double_attr_arg {
     }
 }
 
+#[test]
+fn test_caller() {
+    #[derive(Template)]
+    #[template(
+        source = r#"
+{%- macro test() -%}
+{{- caller() -}}
+{%- endmacro -%}
+{%- call() test() -%}
+nested
+{%- endcall -%}
+"#,
+        ext = "html"
+    )]
+    struct CallerEmpty {}
+    let x = CallerEmpty {};
+    assert_eq!(x.render().unwrap(), "nested");
+}
+
+#[test]
+fn test_caller_struct() {
+    struct TestInput<'a> {
+        a: &'a str, 
+        b: &'a str 
+    }
+    #[derive(Template)]
+    #[template(
+        source = r#"
+{%- macro test(a) -%}
+{{caller(a)}}
+{%- endmacro -%}
+{%- call(value) test(a) -%}
+a: {{value.a}}
+b: {{value.b}}
+{%- endcall -%}
+"#,
+        ext = "txt"
+    )]
+    struct Tmpl<'a> {
+        a: TestInput<'a>
+    }
+    let x = Tmpl {
+        a: TestInput {
+            a: "one",
+            b: "two"
+        }
+    };
+    assert_eq!(x.render().unwrap(), "a: one\nb: two");
+}
+
+#[test]
+fn test_caller_args() {
+    #[derive(Template)]
+    #[template(
+        source = r#"
+{%- macro test() -%}
+{{-  caller("test") -}}
+{{-  caller("test") -}}
```

Comment:
Maybe replace `"test"` with `1` for this one? Like that we cover integer literals as well. :)